onboarding/scaffolding.py

```diff
@@ -0,0 +1,250 @@
+import json
+from pathlib import Path
+from project.util.session_mutation import SessionFormMutation
+from typing import Optional, Tuple
+from django.contrib.gis.geos import GEOSGeometry, Point
+import graphene
+from graphql import ResolveInfo
+import pydantic
+
+from findhelp.models import union_geometries
+
+
+# This should change whenever our scaffolding model's fields change in a
+# backwards incompatible way.
+VERSION = "1"
+
+# Note that this has the word 'norent' in it; this is b/c this structure
+# started out as NoRent-specific data but evolved to be useful elsewhere.
+# We're not renaming this session key b/c we don't want to break existing
+# session data.
+SCAFFOLDING_SESSION_KEY = f"norent_scaffolding_v{VERSION}"
+
+NYC_CITIES = [
+    "nyc",
+    "new york city",
+    "new york",
+    "ny",
+    "manhattan",
+    "queens",
+    "brooklyn",
+    "staten island",
+    "bronx",
+    "the bronx",
+]
+
+BBOUNDS_PATH = Path("findhelp") / "data" / "Borough-Boundaries.geojson"
+
+_nyc_bounds: Optional[GEOSGeometry] = None
+
+
+def is_city_name_in_nyc(city: str) -> bool:
+    parts = city.split("/")
+    for part in parts:
+        if part.strip().lower() in NYC_CITIES:
+            return True
+    return False
+
+
+class OnboardingScaffolding(pydantic.BaseModel):
+    """
+    This is scaffolding we have in place of an actual
+    Django Model (or collection of models), for use while
+    a user is onboarding onto our platform without
+    having had made an account yet.
+    """
+
+    first_name: str = ""
+
+    last_name: str = ""
+
+    preferred_first_name: str = ""
+
+    # e.g. "666 FIFTH AVENUE"
+    street: str = ""
+
+    city: str = ""
+
+    # e.g. "NY"
+    state: str = ""
+
+    # If in NYC, the borough code, e.g. "STATEN_ISLAND".
+    borough: str = ""
+
+    # Whether or not we verified that the user's address was verified
+    # on the server-side.
+    address_verified: bool = False
+
+    # e.g. (-73.9496, 40.6501)
+    lnglat: Optional[Tuple[float, float]] = None
+
+    zip_code: str = ""
+
+    apt_number: Optional[str] = None
+
+    email: str = ""
+
+    phone_number: str = ""
+
+    has_landlord_email_address: Optional[bool] = None
+
+    has_landlord_mailing_address: Optional[bool] = None
```

Comment:
One somewhat annoying thing I'm realizing is that these two fields aren't actually used in onboarding--they're used _after_ account creation to remember what kinds of landlord information the user said they had.

I think that this was partly done because we originally stored _everything_ in `NorentScaffolding` (it wasn't actually just for onboarding, it was to mock out storage for the whole NoRent flow while we developed it), but I kept it around because it didn't really feel like "permanent" information that needed to be stored in a model, but rather in "short-term memory", since it would only be used to remember what to ask the user in immediate subsequent steps.

This makes me wonder if `OnboardingScaffolding` should really be called `ShortTermMemory` or something.  Oy.

Reply:
Eh, I feel like 'remembering info to ask the user about in subsequent steps' is a valid 'onboarding' use case. I like `OnboardingScaffolding`